lib/rets/client.rb

```diff
@@ -278,11 +278,28 @@ def metadata
     def retrieve_metadata
       raw_metadata = {}
       Metadata::METADATA_TYPES.each {|type|
-        raw_metadata[type] = retrieve_metadata_type(type)
+        raw_metadata[type] = filter_metadata(type, retrieve_metadata_type(type))
       }
       raw_metadata
     end
 
+    def filter_metadata(metadata_type, xml)
+      return xml unless @options[:metadata_filter]
+
+      types_to_filter = @options[:metadata_filter][:types_to_filter]
```

Comment:
At this point the metadata is split into:

`SYSTEM RESOURCE CLASS TABLE LOOKUP LOOKUP_TYPE OBJECT`

It doesn't make sense to filter SYSTEM by resource.

Resource is structured differently to Class/Table/Lookup/Lookup_Type/Object.
Each entry for those metadata types has a `Resource` attribute whereas the entries in for the  Resource metadata type has something akin to a `Name` attribute and needs a diffferent code path to filter.